commons/zenoh-util/src/std_only/net/mod.rs

```diff
@@ -412,8 +424,8 @@ pub fn get_interface_names_by_addr(addr: IpAddr) -> ZResult<Vec<String>> {
     }
 }
 
-pub fn get_ipv4_ipaddrs() -> Vec<IpAddr> {
-    get_local_addresses()
+pub fn get_ipv4_ipaddrs(interface: Option<String>) -> Vec<IpAddr> {
```

Comment:
It seems this function does not need to take ownership of the `String`, maybe pass `Option<&str>`?